packages/client/src/containers/RelatedItems/RelatedItems.Container.js

```diff
@@ -1,33 +1,74 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import ProductCard from '../../components/ProductCard/ProductCard.component';
 import { apiURL } from '../../apiURL';
+import { ViewBackCarousel } from '../../components/ViewCarousel/ViewBackCarousel/ViewBackCarousel';
+import { ViewAfterCarousel } from '../../components/ViewCarousel/ViewForwadCarousel/ViewAfterCarousel';
 import './RelatedItems.styles.css';
 
 export const RelatedItems = ({ category }) => {
   const [categoryProducts, setCategoryProducts] = useState([]);
+  const [offSet, setOffset] = useState(0);
+  const [dataFlag, setDataFlag] = useState(true);
 
-  useEffect(() => {
-    const fetchProductsByCategory = async () => {
-      const response = await fetch(`${apiURL()}/products?category=${category}`);
-      const categoryProductData = await response.json();
+  const fetchProductsByCategory = useCallback(async () => {
+    const response = await fetch(
+      `${apiURL()}/products?category=${category}&limit=3&offset=${offSet}`,
+    );
+    const categoryProductData = await response.json();
+
+    if (categoryProductData.data.length < 3) {
+      fetchAllProducts();
+    } else {
+      setDataFlag(true);
+      setCategoryProducts(categoryProductData.data);
+    }
+  }, [category, offSet]);
 
-      setCategoryProducts(categoryProductData);
-    };
+  const fetchAllProducts = async () => {
+    const response = await fetch(`${apiURL()}/products`);
+    const allProductsData = await response.json();
+    const productsData = allProductsData.slice(0, 3);
+    setCategoryProducts(productsData);
+    setDataFlag(false);
+  };
 
+  useEffect(() => {
```

Comment:
I would advice to wrap the logic for the effect inside of the effect, this makes it more clear where these functions belong.

```suggestion
useEffect(() => {
  async function fetchProductsByCategory() {
    const response = await fetch(
      `${apiURL()}/products?category=${category}&limit=3&offset=${offSet}`,
    );
    const { data: products } = await response.json();

    if (products.length < 3) {
      await fetchAllProducts();
    } else {
      setDataFlag(true);
      setCategoryProducts(products);
    }
  }

  async function fetchAllProducts() {
    const response = await fetch(`${apiURL()}/products`);
    const products = await response.json();
    setCategoryProducts(products.slice(0, 3));
    setDataFlag(false);
  }

  fetchProductsByCategory();
}, [category, offset]);
```

Where you might want to use `useCallback` is on the two click handlers below.

Another thing that makes me wonder, why does `api/products` return a different format than `api/products?category`?

Reply:
I didn't understand the comment on useCallback you have mentioned.

Before api/products?category return format  is similar to api/products but I have changed it while doing this component after fetching

Reply:
I understand that you changed it, but my concern is that the change creates a problem. 
The code now has to **know** if it's using the `category` query or not. That's not how things generally are done. The API should keep a consistent return format no matter what query options I send it. 

This is my main concern with this change, that we create a split **inside** the endpoint rather than handling the category logic inside the handler no matter if the query is passed in or not. 

Else, I would argue that these should be two **different** endpoints because their code paths are **not** related and their return format is **not** related.

Reply:
I will change the return format of api/products?category? to same as other endpoints

Reply:
As you suggested like this "Instead, it would be great if your API actually returned a count as well. So the response could look like:{
  status: 200,
  total: 13,
  data: [
    product1,
    product2,
    product3
  ].
}
Can I return when its success